### PR TITLE
feat(graph): Added `not_within` attribute solver for graph checks

### PR DESCRIPTION
**Pull request by bo156**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Added `not_within` attribute solver for graph checks.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/checks_infra/checks_parser.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
     NumberOfWordsGreaterThanAttributeSolver,
     NumberOfWordsGreaterThanOrEqualAttributeSolver,
     NumberOfWordsLessThanAttributeSolver,
-    NumberOfWordsLessThanOrEqualAttributeSolver,
+    NumberOfWordsLessThanOrEqualAttributeSolver, NotWithinAttributeSolver,
```
Comment (gruebel) [inline checkov/common/checks_infra/checks_parser.py:55]: can you move the import directly under the `WithinAttributeSolver` 🙂

File: checkov/common/checks_infra/solvers/attribute_solvers/__init__.py [hunk 2]
```diff
@@ -40,3 +40,4 @@
 from checkov.common.checks_infra.solvers.attribute_solvers.number_of_words_greater_than_or_equal_attribute_solver import NumberOfWordsGreaterThanOrEqualAttributeSolver  # noqa
 from checkov.common.checks_infra.solvers.attribute_solvers.number_of_words_less_than_or_equal_attribute_solver import NumberOfWordsLessThanOrEqualAttributeSolver  # noqa
 from checkov.common.checks_infra.solvers.attribute_solvers.number_of_words_less_than_attribute_solver import NumberOfWordsLessThanAttributeSolver  # noqa
+from checkov.common.checks_infra.solvers.attribute_solvers.not_within_attribute_solver import NotWithinAttributeSolver  # noqa
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/__init__.py:43]: also this one, just move it under `WithinAttributeSolver`

**Review by gruebel — COMMENTED** [review]
nice job 🎄 , please also update the docs here https://github.com/bridgecrewio/checkov/blob/main/docs/3.Custom%20Policies/YAML%20Custom%20Policies.md

File: checkov/common/checks_infra/solvers/attribute_solvers/__init__.py (hunk 2 above)
Comment (bo156) [reply checkov/common/checks_infra/solvers/attribute_solvers/__init__.py:43]: done

**Review by tronxd — APPROVED** [review]
Nice!

**Review by YaaraVerner — APPROVED** [review]

**Review by gruebel — APPROVED** [review]

**Pull request merged**